File: Classes/InfoItem.m
//
//  InfoItem.m
//  FestApp
//
//  Created by Oleg Grenrus on 14/06/14.
//  Copyright (c) 2014 Futurice Oy. All rights reserved.
//

#import "InfoItem.h"

@implementation InfoItem
- (instancetype)initFromJSON:(NSDictionary *)json
{
    self = [super init];
    if (self) {
        _title = json[@"title"];
        _content = json[@"content"];
    }
    return self;
}
@end

File: Classes/InfoItem.swift
//
//  InfoItem.swift
//  FestApp
//
//  Created by Oleg Grenrus on 28/10/14.
//  Copyright (c) 2014 Futurice Oy. All rights reserved.
//

import Foundation

// subclassing NSObject is important
class InfoItem: NSObject {
    let title: String
    let content: String

    init(title: String, content: String) {
        self.title = title
        self.content = content
    }

    // this will generate initWithDictionary objective-c class
    init(dictionary: NSDictionary) {
        title = dictionary["title"] as? String ?? "" // ???
        content = dictionary["content"] as? String ?? ""
    }
}
